python/Ganga/Core/GangaRepository/VStreamer.py

```diff
@@ -299,28 +308,40 @@ def start_element(name, attrs):
             # load a class, make empty object and push it as the current object
             # on the stack
             if name == 'class':
-                try:
-                    cls = allPlugins.find(attrs['category'], attrs['name'])
-                except PluginManagerError as e:
-                    self.errors.append(e)
-                    #self.errors.append('Unknown class: %(name)s'%attrs)
-                    obj = EmptyGangaObject()
-                    # ignore all elemenents until the corresponding ending
-                    # element (</class>) is reached
-                    self.ignore_count = 1
+                if len(self.stack) == 0:
+                    if self.global_obj is not None:
+                        obj = self.global_obj
+                        should_eval = False
+                    else:
+                        should_eval = True
                 else:
-                    version = Version(*[int(v) for v in attrs['version'].split('.')])
-                    if not cls._schema.version.isCompatible(version):
-                        attrs['currversion'] = '%s.%s' % (cls._schema.version.major, cls._schema.version.minor)
-                        self.errors.append(SchemaVersionError('Incompatible schema of %(name)s, repository is %(version)s currently in use is %(currversion)s' % attrs))
+                    should_eval = True
+
+                if should_eval:
+                    try:
+                        cls = allPlugins.find(attrs['category'], attrs['name'])
+                    except PluginManagerError as e:
+                        self.errors.append(e)
+                        #self.errors.append('Unknown class: %(name)s'%attrs)
                         obj = EmptyGangaObject()
                         # ignore all elemenents until the corresponding ending
                         # element (</class>) is reached
                         self.ignore_count = 1
                     else:
-                        # make a new ganga object
-                        obj = super(cls, cls).__new__(cls)
-                        obj.setNodeData({})
+                        version = Version(*[int(v) for v in attrs['version'].split('.')])
+                        if not cls._schema.version.isCompatible(version):
+                            attrs['currversion'] = '%s.%s' % (cls._schema.version.major, cls._schema.version.minor)
+                            self.errors.append(SchemaVersionError('Incompatible schema of %(name)s, repository is %(version)s currently in use is %(currversion)s' % attrs))
+                            obj = EmptyGangaObject()
+                            # ignore all elemenents until the corresponding ending
+                            # element (</class>) is reached
+                            self.ignore_count = 1
+                        else:
+                            # make a new ganga object
+                            obj = cls()
```

Comment:
For the record: this change here (or rather not calling `__init__` until now) was a the source of a slot of the problems with objects not having a `_parent` attribute. I fixed it in the same way in my local branch.